### DISCUSSION TITLE
How to support upgrading GitLab ansible-galaxy role imports?

### DISCUSSION THREAD

**Discussion by cody-reibsome** [renovatebot/renovate#29659] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab - 37.393.0

### Please tell us more about your question or problem

I have a few repositories that include ansible role dependencies via ansible-galaxy as such:

requirements.txt
```
---
- name: role2
  src: gitlab@git.company.io:ansible/roles/role2.git
  version: 3.0.0
  scm: git
```

The git+ssh format supposedly is the only one supported for using GitLab with ansible-galaxy per docs here: https://docs.ansible.com/ansible/latest/galaxy/user_guide.html#installing-multiple-roles-from-a-file

Renovate's git-tags module does not seem able to convert the git+ssh reference into HTTPS in order to retrieve tag info and is instead trying to use ls-remote with the git+ssh reference. Is there a suggested configuration override for doing this?

### Logs (if relevant)

<details><summary>Logs</summary>

```

INFO: Dependency extraction complete (repository=ansible/roles/role1, baseBranch=develop)
       "stats": {
         "managers": {
           "ansible-galaxy": {"fileCount": 2, "depCount": 6},
           "gitlabci-include": {"fileCount": 1, "depCount": 1}
         },
         "total": {"fileCount": 3, "depCount": 7}
       }
DEBUG: Datasource unknown error (repository=ansible/roles/role1)
       "datasource": "git-tags",
       "packageName": "gitlab@git.company.io:ansible/roles/role2.git",
       "err": {
         "task": {
           "commands": [
             "ls-remote",
             "gitlab@git.company.io:ansible/roles/role2.git"
           ],
           "format": "utf-8",
           "parser": "[function]"
         },
         "message": "Could not create directory '/home/ubuntu/.ssh'.\r\nHost key verification failed.\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.\n",

```

</details>

**Comment by secustor** [comment, marked as answer]
You can set an insteadOf rule via a `.ssh/config` or environment variables https://github.com/renovatebot/renovate/blob/main/lib/modules/datasource/git-tags/index.spec.ts#L110-L115

**Reply by cody-reibsome** [reply]
Thanks, this helped point me in the right direction. This ended up working for me:

```
  before_script:
    - 'git config --global url."https://gitlab-ci-token:${CI_JOB_TOKEN}@git.company.io/.insteadOf" gitlab@git.company.io:'

```
